src/App/Tracker/Controller/Hooks/Listeners/AbstractListener.php

```diff
@@ -325,4 +325,149 @@ protected function createStatus(Github $gitHub, TrackerProject $project, $issueN
 			$status->state, $status->targetUrl, $status->description, $status->context
 		);
 	}
+
+	/**
+	 * Set Categories
+	 *
+	 * @param   object       $hookData       Hook data payload
+	 * @param   Logger       $logger         Logger object
+	 * @param   object       $project        Object containing project data
+	 * @param   IssuesTable  $table          Table object
+	 * @param   array        $addCategories  The Categories to add
+	 *
+	 * @return  void
+	 *
+	 * @since   1.0
+	 * @throws  \RuntimeException
+	 */
+	protected function setCategories($hookData, Logger $logger, $project, IssuesTable $table, $addCategories)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueNumber($hookData);
+
+		if ($issueNumber === null)
+		{
+			$message = sprintf('Error retrieving issue number for %s/%s', $project->gh_user, $project->gh_project);
+
+			$logger->error($message);
+
+			throw new \RuntimeException($message);
+		}
+
+		$categoryModel            = new CategoryModel($this->getContainer()->get('db'));
+		$category['issue_id']     = $table->id;
+		$category['modified_by']  = $this->getGithubBotName($project);
+		$category['categories']   = $addCategories;
+		$category['issue_number'] = $issueNumber;
+		$category['project_id']   = $project->project_id;
+
+		$categoryModel->updateCategory($category);
+	}
+
+	/**
+	 * Get Categories
+	 *
+	 * @param   object       $hookData  Hook data payload
+	 * @param   Logger       $logger    Logger object
+	 * @param   object       $project   Object containing project data
+	 * @param   IssuesTable  $table     Table object
+	 *
+	 * @return  array        The object list of the issues.
+	 *
+	 * @since   1.0
+	 * @throws  \RuntimeException
+	 */
+	protected function getCategories($hookData, Logger $logger, $project, IssuesTable $table)
+	{
+		// The Github ID if we have a pull or issue so that method can handle both
+		$issueNumber = $this->getIssueNumber($hookData);
+
+		if ($issueNumber === null)
+		{
+			$message = sprintf('Error retrieving issue number for %s/%s', $project->gh_user, $project->gh_project);
+
+			$logger->error($message);
+
+			throw new \RuntimeException($message);
+		}
+
+		$categoryModel = new CategoryModel($this->getContainer()->get('db'));
+
+		return $categoryModel->getCategories($issueNumber);
```

Comment:
I would change this line to:

```php
$items = $categoryModel->getCategories($table->id);

$categories = [];

foreach ($items as $category)
{
	$categories[] = $category->category_id;
}

return $categories;
```

Thus you will return array of IDs otherwise it will be array of objects.